flixel/sound/FlxSound.hx

```diff
@@ -69,10 +70,16 @@ class FlxSound extends FlxBasic
 	 */
 	public var autoDestroy:Bool;
 	
+	/**
+	 * Signal that is dispatched on sound complete.
+	 */
+	public var onFinish:FlxSignal;
```

Comment:
needs to be read only, lets use `final`

```suggestion
	public final onFinish:FlxSignal;
```

and add `onFinish.removeAll();` to `destroy()` to clear up memory